lib/App/MintTag/Config.pm

```diff
@@ -5,21 +5,28 @@ package App::MintTag::Config;
 use Moo;
 use experimental qw(signatures postderef);
 
+use Path::Tiny qw(path);
 use TOML::Parser;
 
 use App::MintTag::BuildStep;
 use App::MintTag::Remote::GitHub;
 use App::MintTag::Remote::GitLab;
 
-sub from_file ($class, $file) {
+sub from_file ($class, $file, $repo = undef) {
   my $config = TOML::Parser->new->parse_file($file);
 
+  if ($config->{meta}{release_mode}) {
+    die "repo is required for release mode\n" unless $repo;
+    $config = $class->_munge_config_for_release_mode($config, $repo);
+  }
+
   my $remotes = $class->_assemble_remotes($config->{remote});
   my $steps = $class->_assemble_steps($config->{build_steps}, $remotes);
   my $local_conf = $class->_assemble_local_conf($config, $remotes);
 
   return $class->new({
     cfg                => $config,
+    is_release_mode    => $config->{meta}{release_mode},
```

Comment:
I think you want to stick a `!!` in front of that or some weird crap will happen to some future debugger. :)